nixos/modules/services/networking/strongswan.nix

```diff
@@ -114,15 +114,33 @@ in
     };
   };
 
-  config = with cfg; mkIf enable {
+
+  config = with cfg;
+  let
+    # would like to add some path when network manager is enabled
+    secrets2 = cfg.secrets ++ [ "toto" ];
```

Comment:
here in principle it should add conditionnally lib.optional networkmanager_l2tp "/etc/ipsec.d/*.secrets" but whatever I do it never makes it into secret2. For instance here 'toto' is not written to the file. I suspect cfg.secrets to partially support the ++ operator (like it discards its content). The PR should not be merged as long as this is not fixed.